### PR TITLE
Add potion effect type comparator

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
This PR aims to fix a small issue where Skript sometimes has troubles comparing PotionEffectType
This happens with SkBee's potion add event, ex:
```vb
on potion effect added:
    if event-potioneffecttype = nausea:
```
this throws a "can't compare a potion to potion" error.

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

**Review by sovdeeth — CHANGES_REQUESTED** [review]
looks good but needs tests!

File: src/main/java/ch/njol/skript/classes/data/DefaultComparators.java [hunk 1]
```diff
@@ -653,6 +654,14 @@ public boolean supportsOrdering() {
...
 			}
 		});
+
+		// Potion Effect Type
+		Comparators.registerComparator(PotionEffectType.class, PotionEffectType.class, new Comparator<PotionEffectType, PotionEffectType>() {
+			@Override
+			public Relation compare(PotionEffectType effectOne, PotionEffectType effectTwo) {
+				return Relation.get(effectOne.equals(effectTwo));
+			}
+		});
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/classes/data/DefaultComparators.java:664]: I think you can register this like
```suggestion
		Comparators.registerComparator(PotionEffectType.class, PotionEffectType.class, (one, two) -> Relation.get(one.equals(two)));
```

**Review by sovdeeth — APPROVED** [review]
looks good with pickle's

File: src/main/java/ch/njol/skript/classes/data/DefaultComparators.java (hunk 1 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/classes/data/DefaultComparators.java]: done and added, thanks

**Review by APickledWalrus — APPROVED** [review]

**Pull request merged**